dyno-jedis/src/main/java/com/netflix/dyno/jedis/JedisConnectionFactory.java

```diff
@@ -42,47 +38,64 @@
 import com.netflix.dyno.connectionpool.impl.ConnectionContextImpl;
 import com.netflix.dyno.connectionpool.impl.OperationResultImpl;
 
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocketFactory;
+
 public class JedisConnectionFactory implements ConnectionFactory<Jedis> {
 
     private static final org.slf4j.Logger Logger = LoggerFactory.getLogger(JedisConnectionFactory.class);
 
-	private final OperationMonitor opMonitor; 
-	
-	public JedisConnectionFactory(OperationMonitor monitor) {
+	private final OperationMonitor opMonitor;
+    private final SSLSocketFactory sslSocketFactory;
+
+	public JedisConnectionFactory(OperationMonitor monitor, SSLSocketFactory sslSocketFactory) {
 		this.opMonitor = monitor;
+        this.sslSocketFactory = sslSocketFactory;
 	}
-	
+
 	@Override
 	public Connection<Jedis> createConnection(HostConnectionPool<Jedis> pool, ConnectionObservor connectionObservor)
 			throws DynoConnectException, ThrottledException {
-		
+
 		return new JedisConnection(pool);
 	}
 
 	public class JedisConnection implements Connection<Jedis> {
 
 		private final HostConnectionPool<Jedis> hostPool;
-		private final Jedis jedisClient; 
+		private final Jedis jedisClient;
 		private final ConnectionContextImpl context = new ConnectionContextImpl();
-		
+
 		private DynoConnectException lastDynoException;
-		
+
 		public JedisConnection(HostConnectionPool<Jedis> hostPool) {
 			this.hostPool = hostPool;
 			Host host = hostPool.getHost();
-			jedisClient = new Jedis(host.getHostAddress(), host.getPort(), hostPool.getConnectionTimeout(),
-					hostPool.getSocketTimeout());
+
+			if(sslSocketFactory == null)
```

Comment:
hmm what does it mean ? generally if you pass me ```sslSocketFactory``` via builder I will further pass it to Jedis constructor - if not, then ```sslSocketFactory``` will be null, so just non-ssl-aware constructor will be used :)